### PR TITLE
Octal to Binary algo Added

### PR DESCRIPTION
**Pull request by BamaCharanChhandogi**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in their comments that points to Wikipedia or another similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by BamaCharanChhandogi** [comment]
Hey, @raklaptudirm Can you check?

File: Conversions/test/OctToBinary.test.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { octalToBinary } from '../OctToBinary'
+
+test('The Binary representation of Octal number 46 is 100110', () => {
```
Comment (appgurueu) [inline Conversions/test/OctToBinary.test.js:3]: Please use `each` and one `describe` block instead.

File: Conversions/OctToBinary.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    }
+    binary += binDigit // concatenate the binary digit to the binary number
+  }
+  binary = binary.replace(/^0+/, '')
+  return binary
+}
+
+export { octalToBinary }
+
+// > octalToBinary(46)
```
Comment (appgurueu) [inline Conversions/OctToBinary.js:17]: Please add a proper doc comment rather than leaving example code here.

File: Conversions/OctToBinary.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  let binary = '' // an empty string to store the binary number
+  for (let i = 0; i < octal.length; i++) { // loop through each octal digit
+    const digit = parseInt(octal[i], 8) // convert the octal digit to a decimal digit
+    let binDigit = digit.toString(2) // convert the decimal digit to a binary digit
+    while (binDigit.length < 3) { // pad the binary digit with zeros if needed
+      binDigit = '0' + binDigit
+    }
+    binary += binDigit // concatenate the binary digit to the binary number
+  }
+  binary = binary.replace(/^0+/, '')
```
Comment (appgurueu) [inline Conversions/OctToBinary.js:11]: Could directly do `return binary.replace(...)`

File: Conversions/OctToBinary.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+function octalToBinary (octal) {
+  let binary = '' // an empty string to store the binary number
+  for (let i = 0; i < octal.length; i++) { // loop through each octal digit
```
Comment (appgurueu) [inline Conversions/OctToBinary.js:3]: This could be done more elegantly by `map`ing the octal digits to 3 bits each, then `join`ing the resulting array. This avoids the potentially inefficient (O(n²) assuming it isn't optimized out) repeated string concatenation you're doing here.

File: Conversions/test/OctToBinary.test.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { octalToBinary } from '../OctToBinary'
```
Comment (captain-hoity-toity) [inline Conversions/test/OctToBinary.test.js:1]: # Code Review

Overall, the code looks clean and well-organized. However, there are a few small improvements that could be made.

1. **Test Descriptions:** The descriptions of the second and third tests mention 'Decimal' instead of 'Binary'. This could be misleading for anyone reading the tests. I recommend changing 'Decimal' to 'Binary' to accurately reflect what the test is doing.

javascript
test('The Binary representation of Octal number 32 is 11010', () => {
  const res = octalToBinary('32')
  expect(res).toBe('11010')
})

test('The Binary representation of Octal number 17 is 1111', () => {
  const res = octalToBinary('17')
  expect(res).toBe('1111')
})


2. **Consistent Naming:** The function name octalToBinary is in camelCase, which is good, but it's important to use the same casing for all function names to maintain consistency. If there are other functions in your code, make sure they also use camelCase.

3. **Error Handling:** The octalToBinary function does not currently handle errors. What if the input is not a valid octal number? Adding error handling would make the function more robust.

javascript
function octalToBinary(octal) {
  if (!/^([0-7]+)$/.test(octal)) {
    throw new Error('Invalid octal number');
  }
  // rest of the function
}


4. **Test Coverage:** Consider adding more tests to cover edge cases, such as an empty string, a string with non-octal characters, or a string with leading zeros.

Comment (appgurueu) [reply Conversions/test/OctToBinary.test.js:1]: 1. The tests should be rewritten to use `each` first, then the two second test titles will be dropped anyways.
2. Is moot.
3. Error handling is not needed. Adding it is fine though.
4. More test coverage is not needed, but also fine.

Please don't dump AI-generated nonsense reviews.
